### PR TITLE
Pass spec pythonpath directly to spectre

### PR DESCRIPTION
**Pull request by guilara**
## Proposed changes

Since the spec exporter might be a bit difficult to set up, I would like to use the spec python scripts independently of it (still useful for using the pipeline). Could we pass the path directly if it fails to find spec like this?

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: cmake/SetupSpec.cmake [hunk 1]
```diff
@@ -4,6 +4,12 @@
 find_package(SpEC)
 
 if (NOT SpEC_FOUND)
+  # Pass `SPEC_PYTHONPATH` to cmake to make SpEC scripts available in Python
+  # independently of the SpEC exporter. These can be used until we have ported
+  # them to SpECTRE.
+  if (SPEC_PYTHONPATH)
+    set(PYTHONPATH "${SPEC_PYTHONPATH}:${PYTHONPATH}")
```
Comment (nilsvu) [inline cmake/SetupSpec.cmake]: I suggest you just move up the line that sets the pythonpath so it's always set, like this:

```cmake
find_package(SpEC)

# Make SpEC scripts available in Python. These can be used until we have ported
# them to SpECTRE.
if (SPEC_ROOT)
  set(PYTHONPATH "${SPEC_ROOT}/Support/Python:${PYTHONPATH}")
endif()

# That's it. No need to do anything else in this file.
```

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: cmake/SetupSpec.cmake (hunk 1 above)
Comment (guilara) [reply cmake/SetupSpec.cmake]: Oh, right. That's simpler. Done.

File: cmake/SetupSpec.cmake [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
 
-if (NOT SpEC_FOUND)
-  return()
-endif()
-
 # Make SpEC scripts available in Python. These can be used until we have ported
 # them to SpECTRE.
 set(PYTHONPATH "${SPEC_ROOT}/Support/Python:${PYTHONPATH}")
 
+if (NOT SpEC_FOUND)
```
Comment (nilsvu) [inline cmake/SetupSpec.cmake:12]: You have to wrap the line above in `if (SPEC_ROOT)` though

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: cmake/SetupSpec.cmake (hunk 2 above)
Comment (guilara) [reply cmake/SetupSpec.cmake:12]: Oops. Done

**Review by nilsvu — APPROVED** [review]

**Pull request merged**